### PR TITLE
JBPM-7252 - Upgrade camel version to 2.21.0

### PR DESCRIPTION
**Pull request by tsurdilo**

### CODE REVIEW

**Comment by tsurdilo** [comment]
depends on https://github.com/kiegroup/droolsjbpm-build-bootstrap/pull/752

File: camel-workitem/pom.xml [hunk 1]
```diff
@@ -16,6 +16,21 @@
...
+    <maven.jdbc.db.name/>
+    <maven.jdbc.db.port/>
+    <maven.jdbc.db.server/>
+    <maven.jdbc.driver.jar/>
+    <maven.jdbc.username>sa</maven.jdbc.username>
+    <maven.jdbc.password>sasa</maven.jdbc.password>
+    <maven.jdbc.url>jdbc:h2:tcp://localhost/${project.basedir}/target/jbpm-test;MVCC=TRUE</maven.jdbc.url>
+    <maven.jdbc.schema>public</maven.jdbc.schema>
+  </properties>
+
```
Comment (MarianMacik) [inline camel-workitem/pom.xml]: Just curious why we need database with the upgrade, otherwise looks good.

Comment (tsurdilo) [reply camel-workitem/pom.xml]: it is for the camel-sql tests. i can do it as part of a separate pr if you prefer it. np

Comment (MarianMacik) [reply camel-workitem/pom.xml]: If they will be added right after, I don't see any issue with having it prepared.

**Comment by tsurdilo** [comment]
this can be added regardless of the camel version upgrade and is needed to be able to build the project locally (the sql test fixes) so I will go ahead and merge it

**Pull request merged**